src/Compilers/CSharp/Portable/Binder/Binder_Conversions.cs

```diff
@@ -132,11 +132,9 @@ internal partial class Binder
             }
 
             ConstantValue constantValue = this.FoldConstantConversion(syntax, source, conversion, destination, diagnostics);
-            if (conversion.Kind == ConversionKind.DefaultOrNullLiteral && source.Kind == BoundKind.DefaultExpression)
+            if (conversion.Kind == ConversionKind.DefaultLiteral)
             {
-                var defaultExpression = (BoundDefaultExpression)source;
-                Debug.Assert(defaultExpression.TargetType == null);
-                source = defaultExpression.Update(targetType: null, constantValue, destination);
+                return ConvertDefaultLiteral(syntax, (BoundDefaultExpression)source, destination, isCast, conversionGroupOpt, constantValue);
```

Comment:
>return ConvertDefaultLiteral(syntax, (BoundDefaultExpression)source, destination, isCast, conversionGroupOpt, constantValue); [](start = 16, length = 125)

Why couldn't we simply replace the ```source``` with BoundConverted... as the old code used to do?